packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -136,6 +138,34 @@ class StripeHelper {
     );
   }
 
+  /**
+   * Fetch a customer for the record from Stripe based on email.
+   *
+   * @param {string} uid Firefox Account Uid
+   * @param {string} email Firefox Account Email
+   * @param {string[]} [expand] Additional fields to expand in the
+   *                           Stripe call.
+   * @returns {Promise<Customer|void>} Customer if exists in the system.
+   */
+  async fetchCustomer(uid, email, expand) {
+    const customerResponse = await this.stripe.customers
+      .list({ email, expand })
+      .autoPagingToArray({ limit: 20 });
+    if (customerResponse.length === 0) {
+      return;
+    }
+    const customer = customerResponse[0];
+
+    if (customer.metadata.userid !== uid) {
```

Comment:
I suppose in the rare case where the caller passes in `undefined` for the uid and there's no userid in the metadata, this guard will be bypassed.

Reply:
It's my hope that we will start moving to TypeScript sooner rather than later, and this does have JSDoc typing on it for use with typechecking in JS.